src/java/org/apache/cassandra/index/sai/disk/v2/V2VectorIndexSearcher.java

```diff
@@ -243,6 +264,36 @@ private BitsOrPostingList bitsOrPostingListForKeyRange(QueryContext context, Abs
         }
     }
 
+    private int[] findApproximateMatches(QueryContext context, int limit, long minSSTableRowId, long maxSSTableRowId, PrimaryKeyMap primaryKeyMap, OrdinalsView ordinalsView, NodeSimilarity.ApproximateScoreFunction scoreFunction) throws IOException
+    {
+        PriorityQueue<Pair<Integer, Float>> pq = new PriorityQueue<>(limit + 1, Comparator.comparing(Pair::right));
+
+        int[] postings;
+        for (long sstableRowId = minSSTableRowId; sstableRowId <= maxSSTableRowId; sstableRowId++)
```

Comment:
this is very similar to code elsewhere, should we extract it into a rowid iterator?

Reply:
(not a rhetorical question, possibly the answer is "no we should not")

Reply:
Streams work ok in this case